File: python/with_java/programmers/level 2/k진수에서 소수 개수 구하기.py
def make_candidate(n, k):
    nums = []
    while n != 0:
        nums.append(n % k)
        n = n // k
    nums.reverse()
    result = "".join(map(str, nums))
    result = result.split("0")
    return result

def check_prime(num):
    result = True
    for n in range(2, (int)(num ** 0.5)+1):
        if num % n == 0:
            result = False
            break
    return result

def solution(n, k):
    nums = make_candidate(n, k)
    answer = 0
    for num in nums:
        if len(num) == 0:
            continue
        num = int(num)
        if num <= 1:
            continue
        if check_prime(num):
            answer += 1
            
    return answer

File: java/src/main/java/ps_traning/programmers/level_2/k진수에서_소수_개수_구하기.java
package ps_traning.programmers.level_2;

public class k진수에서_소수_개수_구하기 {
    public int solution(int n, int k) {
        int answer = 0;
        String[] nums = Integer.toString(n, k).split("0");
        Loop:
        for (String strNum : nums) {
            if (strNum.length() == 0) continue;
            long num = Long.valueOf(strNum);
            if (num <= 1) continue;
            for (long i = 2; i <= Math.sqrt(num); i++) {
                if (num % i == 0) {
                    continue Loop;
                }
            }
            answer++;
        }
        return answer;
    }
}
